src/briefcase/commands/new.py

```diff
@@ -427,29 +450,63 @@ def build_app_context(self):
             "bundle": bundle,
             "url": url,
             "license": project_license,
-            "gui_framework": (gui_framework.split())[0],
         }
 
+    def build_gui_context(self, context: dict[str, str]) -> dict[str, str]:
+        """Build context specific to the GUI toolkit."""
+        bootstraps = get_gui_bootstraps()
+        bootstrap_choices = [
+            "Toga",
+            "PySide2       (does not support iOS/Android deployment)",
+            "PySide6       (does not support iOS/Android deployment)",
+            "PursuedPyBear (does not support iOS/Android deployment)",
+            "Pygame        (does not support iOS/Android deployment)",
+        ]
+        builtin_bootstraps = [n.split(" ")[0] for n in bootstrap_choices]
+        bootstrap_choices += [f for f in bootstraps if f not in builtin_bootstraps]
+        bootstrap_choices.append("None")
+
+        bootstrap_choice = self.input_select(
+            intro="""
+What GUI toolkit do you want to use for this project?""",
+            variable="GUI framework",
+            options=bootstrap_choices,
+        )
+
+        gui_context = {}
+
+        if bootstrap_choice != "None":
+            try:
+                bootstrap_class = bootstraps[bootstrap_choice]
+            except KeyError:
+                bootstrap_class = bootstraps[bootstrap_choice.split(" ")[0]]
```

Comment:
"There must be a better way"

I didn't _want_ to do a bunch of refactoring here....so, I forced all this to work with what existed and behave the way it did before. There's just a bunch of cascading effects if doesn't behave like before.

I think a big improvement would be string matching on the input choice; so, instead of requiring the input to be the index of the choice, allow an input of "To" to match on Toga. This is obviously non-trivial and I decided not to bite it off here.

That said, I'm open to ideas for improving.